go/adbc/driver/flightsql/flightsql_adbc.go

```diff
@@ -204,19 +227,16 @@ func (d *database) SetOptions(cnOptions map[string]string) error {
 }
 
 type bearerAuthMiddleware struct {
-	token string
+	hdrs metadata.MD
 }
 
 func (b *bearerAuthMiddleware) StartCall(ctx context.Context) context.Context {
-	if b.token != "" {
-		return metadata.AppendToOutgoingContext(ctx, "authorization", b.token)
-	}
-
-	return ctx
+	md, _ := metadata.FromOutgoingContext(ctx)
+	return metadata.NewOutgoingContext(ctx, metadata.Join(md, b.hdrs))
 }
 
 func getFlightClient(ctx context.Context, loc string, d *database) (*flightsql.Client, error) {
-	authMiddle := &bearerAuthMiddleware{}
+	authMiddle := &bearerAuthMiddleware{hdrs: d.hdrs.Copy()}
```

Comment:
Copying means that setting a header after construction won't propagate to children...maybe that's desirable

Reply:
(Well, we also don't want changes in the children to propagate to the parent.)

Reply:
So, if you look at the way it's implemented, the copy here just makes it so that changes to the *database* header options won't propagate to children. But changes to the connection's headers *will* propagate to child statements (which is what you expected in the dropped C++ tests). My original implementation also did a copy from the connection's headers to a statement, but I changed it back when I implemented the tests copying the ones from the dropped C++ impl

Reply:
Yeah, I'm just thinking whether this should work:

1. Set up database
2. Set up connection
3. Add a header at the database level
4. Does the connection in step 2 reflect the new header?

I think it's OK if it doesn't so long as it's documented (so, copy on construct always) - keeps the implementation clean and avoids a bunch of shared state

Reply:
If you look at the old C++ implementation, statements had a strong reference to connections and connections had a strong reference to databases, so each call walked the chain of references and built up the final dictionary of headers dynamically - that's what was behind the behavior in those C++ tests you saw

Reply:
I like not having it reflect the new header. whatever headers exist at the moment of creation should propagate, but subsequent changes shouldn't propagate (same for connections and statements).

The updated commit and tests demonstrate this and show it.